aea/configurations/components.py

```diff
@@ -0,0 +1,178 @@
+        return self._is_vendor
+
+    @property
+    def prefix_import_path(self):
+        """Get the prefix import path for this component."""
+        if self.is_vendor:
+            return "packages.{}.{}.{}".format(
+                self.public_id.author,
+                self.component_type.to_plural(),
+                self.public_id.name,
+            )
+        else:
+            return "packages.{}.{}".format(
+                self.component_type.to_plural(), self.public_id.name
+            )
+
+    @property
+    def component_id(self) -> ComponentId:
+        """Ge the package id."""
+        return self._configuration.component_id
+
+    @property
+    def public_id(self) -> PublicId:
+        """Get the public id."""
+        return self._configuration.public_id
+
+    @property
+    def configuration(self) -> ComponentConfiguration:
+        """Get the component configuration."""
+        return self._configuration
+
+    @property
+    def directory(self) -> Optional[Path]:
+        """Get the directory, or None if the component is not associated with any directory."""
+        return self._directory
+
+    def setup(self, *args, **kwargs):
+        """
+        Set the component up.
+
+        This method is called by the framework before running the agent.
+        The implementation varies depending on the type of component.
+        Please check the concrete component classes.
+        """
+
+    @classmethod
+    def _load_connection(cls, configuration: ConnectionConfig, directory: Path):
+        """Load a connection object from a directory."""
+        connection_module = load_module(
+            "connection_module", directory / "connection.py"
+        )
+        classes = inspect.getmembers(connection_module, inspect.isclass)
+        connection_classes = list(
+            filter(lambda x: re.match("\\w+Connection", x[0]), classes)
+        )
+        name_to_class = dict(connection_classes)
+        connection_class_name = cast(str, configuration.class_name)
+        logger.debug("Processing connection {}".format(connection_class_name))
+        connection_class = name_to_class.get(connection_class_name, None)
+        if connection_class is None:
+            raise ValueError(
+                "Connection class '{}' not found.".format(connection_class_name)
+            )
+
+        # TODO address?
+        connection = connection_class.from_config(
+            "address", connection_configuration=configuration
+        )
+        connection._configuration = configuration
+        return connection
+
+    @classmethod
+    def load_from_directory(
+        cls, component_type: ComponentType, directory: Path
+    ) -> "Component":
+        """Load a component from the directory."""
+        configuration = ComponentConfiguration.load(component_type, directory)
+        if component_type == ComponentType.CONNECTION:
+            # the connection case is an exception,
```

Comment:
good point. I think we did that because of the gym connection. Should be solvable.

Reply:
I was thinking of using the same paradigm we used for `Protocol` and `Serialization`.
For connection, we could have something similar but with `Channel`.

Reply:
Let's use `setup` and ask developer to implement it. Then call it here.